### PR TITLE
Topics 1, 2 and 3 (updated topic 0)

### PR DESCRIPTION
**Pull request by victor20m**
-Added suggested changes by @vallejolanGlober  to topic 0.
-Added topic 1
-Added topic 2 along with documentation and implemented the code of topic 3.

### CODE REVIEW

File: Topic_0/.metadata/.log [hunk 1]
```diff
@@ -0,0 +1,206 @@
+!SESSION 2018-09-27 13:54:50.341 -----------------------------------------------
```
Comment (vallejolanGlober) [inline Topic_0/.metadata/.log:1]: you shouldn't have versioned this file. Add it to .ignore

File: Topic_0/.metadata/.plugins/org.eclipse.core.resources/.history/5e/70c80b7076c200181749aea7f0ae50b9 [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+public class DatabaseProxy implements DataRequest {
+
+	private User registeredUser;
+	private Database mysqlDatabase;
+
+	/*
+	 * When a DatabaseProxy instance is created, it receives the user making the
+	 * requests
+	 */
+
```
Comment (vallejolanGlober) [inline Topic_0/.metadata/.plugins/org.eclipse.core.resources/.history/5e/70c80b7076c200181749aea7f0ae50b9:12]: this empty line shouldn't be here

**Review by vallejolanGlober — CHANGES_REQUESTED** [review]
please, this PR is impossible to review, ignore all unnecessary files and if necessary close the PR and create a new one.

**Comment by jdavisonc** [comment]
Hi Victor, I'm seeing files like `.plugins/org.eclipse.core.resources/.history/e7/f03d9f677dc200181749aea7f0ae50b9`, can we ignore them or remove them from the PR?

**Pull request closed without merging**